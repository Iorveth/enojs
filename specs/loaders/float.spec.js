const eno = require('enojs');

const examples = {
  '42':       42.0,
  '-42':      -42.0,
  '42.0':     42.0,
  '42,0':     false,
  '4 2.0':    false,
  'fortytwo': false
};

describe('float loader', () => {
  for(let [value, result] of Object.entries(examples)) {
    describe(value, () => {
      const document = eno.parse(`value: ${value}`);

      if(result === false) {
        it('throws an error', () => {
          expect(() => document.float('value')).toThrowErrorMatchingSnapshot();
        });
      } else {
        it(`returns ${result}`, () => {
          expect(document.float('value')).toBe(result);
        });
      }
    });
  }
});
