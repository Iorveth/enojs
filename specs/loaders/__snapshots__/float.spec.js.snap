// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`float 4 2.0 throws an error 1`] = `
"'float' must contain a decimal number, for instance '13.0', '-9.159' or '42'.

   Line | Content
 >    1 | float: 4 2.0
"
`;

exports[`float 42,0 throws an error 1`] = `
"'float' must contain a decimal number, for instance '13.0', '-9.159' or '42'.

   Line | Content
 >    1 | float: 42,0
"
`;

exports[`float fortytwo dot zero throws an error 1`] = `
"'float' must contain a decimal number, for instance '13.0', '-9.159' or '42'.

   Line | Content
 >    1 | float: fortytwo dot zero
"
`;
